src/backend/distributed/commands/trigger.c

```diff
@@ -203,45 +255,605 @@ get_relation_trigger_oid_compat(HeapTuple heapTuple)
+
+	List *targetObjectList = dropTriggerStmt->objects;
+	List *triggerObjectNameList = linitial(targetObjectList);
+
+	/*
+	 * The name list that identifies the trigger to be dropped looks like:
+	 * [catalogName, schemaName, relationName, triggerName], where, the first
+	 * two elements are optional. We should take all elements except the
+	 * triggerName to create the range var object that defines the owner
+	 * relation.
+	 */
+	int relationNameListLength = list_length(triggerObjectNameList) - 1;
+	List *relationNameList = list_truncate(list_copy(triggerObjectNameList),
+										   relationNameListLength);
+
+	return makeRangeVarFromNameList(relationNameList);
+}
+
+
+/*
+ * DropTriggerEventExtendNames extends relation name and trigger name with
+ * shardId, and sets schema name in given DropStmt by recreating "objects"
+ * list.
+ */
+void
+DropTriggerEventExtendNames(DropStmt *dropTriggerStmt, char *schemaName, uint64 shardId)
+{
+	Assert(dropTriggerStmt->removeType == OBJECT_TRIGGER);
+
+	char *triggerName = NULL;
+	char *relationName = NULL;
+	ExtractDropStmtTriggerAndRelationName(dropTriggerStmt, &triggerName, &relationName);
+
+	AppendShardIdToName(&triggerName, shardId);
+	Value *triggerNameValue = makeString(triggerName);
+
+	AppendShardIdToName(&relationName, shardId);
+	Value *relationNameValue = makeString(relationName);
+
+	Value *schemaNameValue = makeString(pstrdup(schemaName));
+
+	List *shardTriggerNameList =
+		list_make3(schemaNameValue, relationNameValue, triggerNameValue);
+	dropTriggerStmt->objects = list_make1(shardTriggerNameList);
+}
+
+
+/*
+ * ExtractDropStmtTriggerAndRelationName extracts triggerName and relationName
+ * from given dropTriggerStmt if arguments are passed as non-null pointers.
+ */
+static void
+ExtractDropStmtTriggerAndRelationName(DropStmt *dropTriggerStmt, char **triggerName,
+									  char **relationName)
+{
+	ErrorIfDropStmtDropsMultipleTriggers(dropTriggerStmt);
+
+	List *targetObjectList = dropTriggerStmt->objects;
+	List *triggerObjectNameList = linitial(targetObjectList);
+	int objectNameListLength = list_length(triggerObjectNameList);
+
+	if (triggerName != NULL)
+	{
+		int triggerNameindex = objectNameListLength - 1;
+		*triggerName = strVal(list_nth(triggerObjectNameList, triggerNameindex));
+	}
+
+	if (relationName != NULL)
+	{
+		int relationNameIndex = objectNameListLength - 2;
+		*relationName = strVal(list_nth(triggerObjectNameList, relationNameIndex));
+	}
+}
+
+
+/*
+ * ErrorIfDropStmtDropsMultipleTriggers errors out if given drop trigger
+ * command drops more than one triggers. Actually, this can't be the case
+ * as postgres doesn't support dropping multiple triggers, but we should
+ * be on the safe side.
+ */
+static void
+ErrorIfDropStmtDropsMultipleTriggers(DropStmt *dropTriggerStmt)
+{
+	List *targetObjectList = dropTriggerStmt->objects;
+	if (list_length(targetObjectList) > 1)
+	{
+		ereport(ERROR, (errmsg("bug: cannot execute DROP TRIGGER command "
```

Comment:
I just followed what we have done in `CalculateUniformHashRangeIndex`. I can't see any specific benefits except telling users that:
* it is not a limitation actually 
* it should be reported to us :)
But of course, the downside of it would be telling that we have bugs :D

So, I'm not much sure if we should or shouldn't put `bug: ` to such error messages.

Reply:
@onderkalaci any comments?